kolibri/deployment/default/settings/base.py

```diff
@@ -126,25 +128,38 @@
 # Database
 # https://docs.djangoproject.com/en/1.9/ref/settings/#databases
 
-DATABASES = {
-    'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(KOLIBRI_HOME, 'db.sqlite3'),
-        'OPTIONS': {
-            'timeout': 100,
+if OPTIONS.get("DATABASE_ENGINE") == "sqlite":
+    DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.sqlite3',
+            'NAME': os.path.join(KOLIBRI_HOME, OPTIONS.get('DATABASE_NAME') or 'db.sqlite3'),
+            'OPTIONS': {
+                'timeout': 100,
+            }
         }
-    },
-}
+    }
+elif OPTIONS.get("DATABASE_ENGINE") == "postgres":
+    DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.postgresql',
+            'NAME': OPTIONS['DATABASE_NAME'],
+            'PASSWORD': OPTIONS.get('DATABASE_PASSWORD') or '',
+            'USER': OPTIONS.get('DATABASE_USER') or '',
+            'HOST': OPTIONS.get('DATABASE_HOST') or '',
+        }
+    }
 
 # Content directories and URLs for channel metadata and content files
 
+_content_root = os.path.join(KOLIBRI_HOME, OPTIONS.get('CONTENT_ROOT') or 'content')
+
 # Directory and URL for storing content databases for channel data
-CONTENT_DATABASE_DIR = os.path.join(KOLIBRI_HOME, 'content', 'databases')
+CONTENT_DATABASE_DIR = os.path.join(_content_root, 'databases')
 if not os.path.exists(CONTENT_DATABASE_DIR):
     os.makedirs(CONTENT_DATABASE_DIR)
 
 # Directory and URL for storing de-duped content files for all channels
-CONTENT_STORAGE_DIR = os.path.join(KOLIBRI_HOME, 'content', 'storage')
+CONTENT_STORAGE_DIR = os.path.join(_content_root, 'storage')
 if not os.path.exists(CONTENT_STORAGE_DIR):
     os.makedirs(CONTENT_STORAGE_DIR)
```

Comment:
These settings remind of our "derived settings" which caused issues in KA Lite. It's very nice to define them in a non-configurable way, but having them sit in the django settings module is perhaps not the right place, since it implies that they can be configured? Maybe they can? But should they?

Reply:
>These settings remind of our "derived settings" which caused issues in KA Lite.

Critical difference here: I prefixed `_content_root` with an underscore so that it's not exported -- i.e. it's not a Django setting, and can't be overridden in Django settings files, so we don't end up with the "I changed this and the derived stuff didn't change" confusion.

>It's very nice to define them in a non-configurable way, but having them sit in the django settings module is perhaps not the right place, since it implies that they can be configured? Maybe they can? But should they?

We will not be recommending to regular end users that they ever use Django settings to configure anything. `CONTENT_STORAGE_DIR` and `CONTENT_DATABASE_DIR` are for a specific "power user" case that is already being utilized by us on our demo servers -- having a single shared `CONTENT_STORAGE_DIR` across demo instances, but separate `CONTENT_DATABASE_DIR` directories, so that we can leverage deduplicated content storage while showing different (but sometimes overlapping) content across demo servers.

Reply:
I guess, could we define these in `conf` instead, as you have with the others? I think this is a nice pattern. I think we can stomach the one time cost of updating our demo servers to use .ini files, no?

Reply:
Nope, didn't want to expose them in the ini file, just thinking it is good to not give multiple paths to define variables. As these are are purely Kolibri config, they don't need to be in django settings at all, and can be defined in conf based on the value you are using here.

Reply:
Ah, but how will they then be overridden independently, for the use case described earlier in this comment thread?

>CONTENT_STORAGE_DIR and CONTENT_DATABASE_DIR are for a specific "power user" case that is already being utilized by us on our demo servers -- having a single shared CONTENT_STORAGE_DIR across demo instances, but separate CONTENT_DATABASE_DIR directories, so that we can leverage deduplicated content storage while showing different (but sometimes overlapping) content across demo servers.

Reply:
I think the main thing I'm unclear on is what "define these in conf instead" means. Could you clarify the proposal?

Reply:
Oh, well, the use case you describe is no really needed, right? It doesn't matter if different demo servers share a data folder, because they don't draw from that folder (the only time this is not the case is on upgrade, but we can  make this behaviour conditional on upgrade from a very old version).

So, let's not allow them to be configured separately, because there is is no need, and just define these derived values in the conf module?

Reply:
The other option is just to keep the database folder completely empty.

Reply:
>we can make this behaviour conditional on upgrade from a very old version

Not quite sure what you mean -- do you mean modify the `.data_version` to trick it into thinking it doesn't need to run the upgrade code even after we upgrade? What if we have other upgrade logic that *should* be run after upgrade?

>The other option is just to keep the database folder completely empty.

Like... with a cron job that runs every 30 mins or something...? We want to be able to update the content on these demo servers regularly (via the UI), and they download stuff into that folder.

Reply:
> we can make this behaviour conditional on upgrade from a very old version

Oh, I see what you mean -- change the checking logic so it only happens when the old version is < some cutoff.

Reply:
I think we can actually solve this on demo servers in a totally different way -- not use settings at all, and just symlink the content/storage directories to a shared location -- I think we actually started using that strategy, in some cases, and it should be sufficient.

So yeah, I can move these out of the settings. In terms of where to put it, I still feel weird putting too much at the top level of `kolibri.utils.conf`, because [that code](https://github.com/learningequality/kolibri/blob/develop/kolibri/utils/conf.py) is all very oriented around module-level variables and functions related to the `kolibri_settings.json` stuff. I'm thinking to move that stuff to its own module, and have the conf module be the place all this stuff is bound together and exposed. May try a pass at that now.

Reply:
Actually, I think the best place to put this will be in https://github.com/learningequality/kolibri/blob/develop/kolibri/content/utils/paths.py

(And then I'll hold off on a big refactor of where things live, until we revisit `kolibri_settings.json` more broadly -- including a rename of that file!)

Reply:
Yes indeed - have always used symlinks for such tasks... the reason for the necessity of this setting is AFAIK just Windows file systems, which don't support symlinks.

Maybe we should also check if our disk space measurement resolves when subdirs are symlinked.

Reply:
@jamalex 

> Critical difference here: I prefixed `_content_root` with an underscore so that it's not exported -- i.e. it's not a Django setting, and can't be overridden in Django settings files, so we don't end up with the "I changed this and the derived stuff didn't change" confusion.

Because `CONTENT_STORAGE_DIR` (not sure if still the name) is a setting that's derived from `_content_root`, the user can still pick a value for `CONTENT_STORAGE_DIR` that's inconsistent with `_content_root`.

So then the problem becomes "the user changed something that was derived from `_content_root`, which we thought was an un-changeable internal value" :)

It becomes a lot better if the variable is renamed `_default_content_root_prefix` :)